server/server_test.go

```diff
@@ -248,7 +241,7 @@ func TestIdleOriginDirect(t *testing.T) {
 	failForwardFn := func(conn net.Conn, originURL *url.URL) {
 		var buf [400]byte
 		chunkedReq(t, &buf, conn, originURL)
-		assert.Contains(t, string(buf[:]), "502 Bad Gateway", "should fail with 502")
```

Comment:
This is just an effect of the new implementation in proxy. If the upstream host disconnects prematurely, our proxy simply disconnects too.